gooresmed/download.py

```diff
@@ -124,6 +124,41 @@ class Download(_DownloadBase):
            ``start`` to the end of the media.
     """
 
+    def _prepare_request(self):
+        """Prepare the contents of an HTTP request.
+
+        .. _sans-I/O: https://sans-io.readthedocs.io/
+
+        This is everything that must be done before a request that doesn't
+        require network I/O (or other I/O). This is based on the `sans-I/O`_
+        philosophy.
+
+        Returns:
+            dict: The headers for the request.
+
+        Raises:
+            ValueError: If the current :class:`Download` has already
+                finished.
+        """
+        if self.finished:
+            raise ValueError('A download can only be used once.')
+
+        headers = {}
+        _add_bytes_range(self.start, self.end, headers)
+        return headers
+
+    def _process_response(self):
+        """Process the response from an HTTP request.
+
+        .. _sans-I/O: https://sans-io.readthedocs.io/
+
+        This is everything that must be done after a request that doesn't
+        require network I/O (or other I/O). This is based on the `sans-I/O`_
+        philosophy.
+        """
+        # Tombstone the current Download so it cannot be used again.
+        self._finished = True
+
     def consume(self, transport):
```

Comment:
Nah, not that, just that they're different primitives.

e.g., you'd have `download.sansio` for all the sans-io pieces then `download.adapters` for the actual classes that implement 'consume'. 

It's up to you. IMO it's borderline overkill because this library will almost always be tied to google auth's transport interface.